docs/typescript.md

```diff
@@ -2,17 +2,47 @@
 
 **_Purpose_**: Establish best practices for writing robust, maintainable, and efficient TypeScript code.
 
-**_Audience_**: Internal and external developers using TypeScript in Metamask.
+**_Audience_**: Internal and external developers using TypeScript to contribute to MetaMask.
 
-## **Type**
+### Implements
 
-By defining expected object structures with interfaces, TypeScript promotes developer intent and unlocks powerful IDE features like accurate completion and context-aware navigation.
+**`implements` keyword**
 
-### **Enums**
+The `implements` keyword enables us to define and enforce interfaces, i.e. strict contracts consisting of expected object and class properties and abstract method signatures.
+Writing an interface to establish the specifications of a class that external code can interact while without being aware of internal implementation details is encouraged as sound OOP development practice.
+Here's an abbreviated example from `@metamask/polling-controller` of an interface being used to define one of our most important constructs.
+```typescript
+export type IPollingController = {
+...
+}
+
+export function AbstractPollingControllerBaseMixin<TBase extends Constructor>(
+    Base: TBase,
+) {
+    abstract class AbstractPollingControllerBase
+        extends Base
+        implements IPollingController
+    { ... }
+    return AbstractPollingControllerBase
+}
+```
+
+The concept of the interface as discussed in this section is not to be confused with interface syntax as opposed to type alias syntax. Note that in the above example, the `IPollingController` interface is defined as a type alias, not using the `interface` keyword.
+
+**Always prefer type aliases over `interface` keyword**
```

Comment:
Nit: is this maybe a section heading rather than just emphasized text?

Reply:
Good catch !! changed !